### PR TITLE
Update r 4.2.2

### PR DESCRIPTION
**Pull request by gerbeldo**
# Description
Update R to fix worker build, which fails due to the R package `nloptr`, which requires nlopt > 2.7, which is only available in ubuntu jammy, which is the ubuntu version used by Rocker starting from R-4.2.2.

This also makes R quiet (by adding the `-q` and `-s` flags to R calls in the dockerfile (fixing the call to conda clean).

Updates renv to the latest version.

Makes trajectory analysis tests a bit more stable by using better datasets and removing unnecessary checks.

Finally, updates the `renv.lock.init` to the state of the current `renv.lock`, to take build times back from 2.5h to 1.5h.

# Details
#### URL to issue
N/A

#### Link to staging deployment URL (or set N/A)
https://ui-german-worker61.scp-staging.biomage.net/

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [x] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: r/renv.lock [hunk 1]
```diff
@@ -2443,13 +2440,20 @@
     },
     "reticulate": {
       "Package": "reticulate",
-      "Version": "1.20",
-      "Source": "Repository",
-      "Repository": "CRAN",
-      "Hash": "30ab0ea8c8d3dd16a3fa06903449bbfb",
+      "Version": "1.26-9000",
```
Comment (gerbeldo) [inline r/renv.lock:2443]: this fixes the miniconda bug.

File: r/setup/sysdeps_build_debian.txt [hunk 2]
```diff
@@ -17,4 +17,5 @@ libgeos-dev
 procps
 libudunits2-dev
 libgdal-dev
-cmake
\ No newline at end of file
+cmake
+libnlopt-dev
```
Comment (gerbeldo) [inline r/setup/sysdeps_build_debian.txt:21]: required by nloptr, which is failing build in previous worker image

File: r/tests/testthat/test-trajectory_analysis.R [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
-    keep_cells_id <- which(!pbmc_small$cells_id %in% filt_cell_id)
-    keep_cells <- names(pbmc_small$cells_id[keep_cells_id])
-    pbmc_small <- subset(pbmc_small, cells = keep_cells)
-  }
-  pbmc_small@misc$gene_annotations <- data.frame(
-    input = row.names(pbmc_small),
-    name = row.names(pbmc_small),
-    row.names = row.names(pbmc_small)
+mock_scdata <- function(){
+  cds <- monocle3::load_a549()
```
Comment (gerbeldo) [inline r/tests/testthat/test-trajectory_analysis.R:2]: use monocle data for trajectory tests. the dataset is a bit bigger which makes trajectory analysis more stable between environments (some tests fail in containers but not in Rstudio due to small numerical differences; using a bigger dataset fixes that)

File: r/tests/testthat/test-trajectory_analysis.R [hunk 4]
```diff
@@ -133,10 +131,6 @@ test_that("runTrajectoryAnalysisStartingNodesTask output has the expected format
   expect_named(root_nodes, c("connectedNodes", "x", "y"))
   expect_type(root_nodes$x[[1]], "double")
   expect_type(root_nodes$y[[1]], "double")
-  expect_type(root_nodes$connectedNodes[[1]], "double")
```
Comment (gerbeldo) [inline r/tests/testthat/test-trajectory_analysis.R:136]: this test is not constant, and breaks often. We have unittests for the function that ensures the type, so these checks here are not really required.

File: r/renv/activate.R [hunk 5]
```diff
@@ -2,7 +2,7 @@
 local({
 
   # the requested version of renv
-  version <- "0.15.5"
+  version <- "0.16.0"
```
Comment (gerbeldo) [inline r/renv/activate.R:5]: changes in this file are automatic, when updating the renv version.

**Review by kafkasl — APPROVED** [review]

**Pull request merged**